### PR TITLE
projectile: autoload `projectile-project-p`

### LINKED ISSUE #5127: spacemacs/helm-persp-switch-project assumes projectile is loaded
The `spacemacs/helm-persp-switch-project` function (bound to `SPC p l` by default) assumes that the `projectile` package is loaded (specifically, it assumes that the `projectile-project-p` function is defined). This means that running `SPC p l` immediately after starting spacemacs yields an error. One solution is to add `(autoload 'projectile-project-p "projectile")` somewhere but I don't know if this is the "correct" solution or where this line should go.

### PR DESCRIPTION
**Pull request by bixuanzju**
Fix #5127

### CODE REVIEW

File: layers/+distribution/spacemacs-base/packages.el [hunk 1]
```diff
@@ -781,6 +781,7 @@ below. Anything else exits."
                projectile-kill-buffers
                projectile-multi-occur
                projectile-project-root
+               projectile-project-p
```
Comment (StreakyCobra) [inline layers/+distribution/spacemacs-base/packages.el]: One line above ;-)

Comment (d12frosted) [inline layers/+distribution/spacemacs-base/packages.el]: `sort-lines` ! :D

Comment (bixuanzju) [inline layers/+distribution/spacemacs-base/packages.el]: Sure thing! Since i am on that, `projectile-grep` is also misplaced.

Comment (StreakyCobra) [inline layers/+distribution/spacemacs-base/packages.el]: :+1:

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
Released in 0.105.14

**Pull request closed without merging**